### Exercise 2

In [17]:
import requests
import json

# Setting  IBM Watson NLU API credentials and endpoint
apikey = 'qXBbo61tWIEYAPWk1fJByjbyCpfY0hZR3jNc6xvXEka9'
url = 'https://api.us-south.natural-language-understanding.watson.cloud.ibm.com/instances/f55ba5ba-4ca9-4ec2-a47b-d72fe26d7fc3/v1/analyze?version=2019-07-12'

# The provided text
text_to_analyze = """Claudine Gay resigned Tuesday as Harvard’s president after weeks of controversy
initially sparked by her testimony at a congressional hearing on antisemitism, and later
focusing on allegations of plagiarism in work she had produced throughout her academic
career. (Gay’s resignation was first reported by the Harvard Crimson.) Plagiarism is a
particularly serious charge in academia: In 2021, the president of the University of South
Carolina, Lt. Gen. Robert L. Caslen Jr., resigned after admitting that he had plagiarized a
commencement speech. Plagiarism includes a broad range of possible offenses, from
improperly placing a citation to claiming credit for another scholar’s research or insight,
and individual institutions develop their own policies for evaluating and responding to
allegations of research misconduct. Gay defended her academic record in an op-ed
published Wednesday in the New York Times, saying that she addressed attribution
issues promptly after learning about them, and that “I have never misrepresented my
research findings, nor have I ever claimed credit for the research of others. Moreover,
the citation errors should not obscure a fundamental truth: I proudly stand by my work
and its impact on the field.”"""

# Setup the request payload
payload = {
    "text": text_to_analyze,
    "features": {
        "sentiment": {}
    }
}

# Making the POST request to the IBM Watson NLU API
response = requests.post(url, auth=('apikey', apikey), headers={'Content-Type': 'application/json'}, data=json.dumps(payload))

# Check if the request was successful and print the response
if response.status_code == 200:
    print(json.dumps(response.json(), indent=2))
else:
    print(f"Request failed with status code {response.status_code}: {response.text}")


{
  "usage": {
    "text_units": 1,
    "text_characters": 1238,
    "features": 1
  },
  "sentiment": {
    "document": {
      "score": -0.358586,
      "mixed": "1",
      "label": "negative"
    }
  },
  "language": "en"
}


## Comparison of Results

### IBM Watson NLU Result:

- **Score**: -0.358586
- **Label**: Negative

### Transformer Model Result 

- **Score**: 0.900300920009613
- **Label**: POSITIVE


## IBM's Philosophy in Sentiment Analysis

IBM Watson identifies complexity in human emotions by detecting not only the dominant sentiment but also mixed sentiments in texts. It provides a detailed view of a text's emotional content through a sentiment score, a label, and an indicator of mixed sentiments.

